Test failure seemingly from Kafka during release workflow ExpositionIT#canConfigureLowestBrokerIdWithPortPerBroker

**Describe the bug**

We saw a test failure during a release workflow.  It appeared to be from Kafka itself, rather than Kroxylicious.
I haven't seen this issue before.  I haven't investigated further.

```
2024-08-20T12:49:04.7868051Z [ERROR] io.kroxylicious.proxy.ExpositionIT.canConfigureLowestBrokerIdWithPortPerBroker(KafkaCluster, Admin) -- Time elapsed: 1.058 s <<< ERROR!
2024-08-20T12:49:04.7870518Z java.lang.IllegalArgumentException: Broker node 0 is not a member of the cluster.
2024-08-20T12:49:04.7872622Z 	at io.kroxylicious.testing.kafka.invm.InVMKafkaCluster.removeBroker(InVMKafkaCluster.java:313)
2024-08-20T12:49:04.7875008Z 	at io.kroxylicious.proxy.ExpositionIT.canConfigureLowestBrokerIdWithPortPerBroker(ExpositionIT.java:514)
2024-08-20T12:49:04.7877107Z 	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
2024-08-20T12:49:04.7878741Z 	at java.base/java.util.ArrayList.forEach(ArrayList.java:1596)
2024-08-20T12:49:04.7880248Z 	at java.base/java.util.ArrayList.forEach(ArrayList.java:1596)
```

**To Reproduce**
Steps to reproduce the behavior:
1. Configure kroxylicious like this ...
2. Run command '....'
3. See error

**Expected behavior**
A clear and concise description of what you expected to happen.

**YAML files and logs**

Attach or copy paste your kroxylicious configuration YAML (with sensitive details
like broker address redacted).
Attach or copy and paste relevant logs.

**Additional context**
Add any other context about the problem here.

[logs_27378716046.zip](https://github.com/user-attachments/files/16676133/logs_27378716046.zip)

--- a/impl/src/main/java/io/kroxylicious/testing/kafka/invm/InVMKafkaCluster.java
+++ b/impl/src/main/java/io/kroxylicious/testing/kafka/invm/InVMKafkaCluster.java
@@ -18,13 +18,14 @@ import java.nio.file.Files;
 import java.nio.file.Path;
 import java.util.Arrays;
 import java.util.Comparator;
-import java.util.HashMap;
 import java.util.HashSet;
 import java.util.List;
 import java.util.Map;
 import java.util.Optional;
 import java.util.Properties;
 import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
 import java.util.concurrent.TimeUnit;
 import java.util.function.Function;
 import java.util.function.IntPredicate;
@@ -75,9 +76,8 @@ public class InVMKafkaCluster implements KafkaCluster, KafkaClusterConfig.KafkaE
 
     /**
      * Map of kafka <code>node.id</code> to {@link Server}.
-     * Protected by lock of {@link InVMKafkaCluster itself.}
      */
-    private final Map<Integer, Server> servers = new HashMap<>();
+    private final ConcurrentMap<Integer, Server> servers = new ConcurrentHashMap<>();
 
     /**
      * Set of kafka <code>node.id</code> that are currently stopped.

